python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -577,47 +576,46 @@ def _set_mpls_route_clients(self: AvdStructuredConfigOverlayProtocol) -> None:
         if self._is_mpls_server is not True:
             return
 
-        for avd_peer in natural_sort(self._avd_overlay_peers):
-            peer_facts = self.shared_utils.get_peer_facts(avd_peer, required=True)
-            if self._is_peer_mpls_client(peer_facts) is not True:
+        for route_reflector_client in natural_sort(self.facts.mpls_route_reflector_clients):
+            if route_reflector_client in self._mpls_route_reflectors:
                 continue
 
-            if self.shared_utils.hostname in peer_facts.get("mpls_route_reflectors", []) and avd_peer not in self._mpls_route_reflectors:
-                neighbor = self._create_neighbor(
-                    get(
-                        peer_facts,
-                        "overlay.peering_address",
-                        required=True,
-                        custom_error_msg=f"switch.overlay.peering_address for {avd_peer} is required.",
-                    ),
-                    avd_peer,
-                    self.inputs.bgp_peer_groups.mpls_overlay_peers.name,
-                    overlay_peering_interface="Loopback0",
-                )
+            peer_facts = self.shared_utils.get_peer_facts(route_reflector_client)
+            if not self._is_peer_mpls_client(peer_facts):
+                continue
 
-                self.structured_config.router_bgp.neighbors.append(neighbor)
+            if not (ip_address := peer_facts.overlay.peering_address):
+                msg = f"Unable to determine the remote IP address to use for the MPLS Route Reflector client '{route_reflector_client}'."
+                raise AristaAvdInvalidInputsError(msg)
+
+            neighbor = self._create_neighbor(
+                ip_address,
+                route_reflector_client,
+                self.inputs.bgp_peer_groups.mpls_overlay_peers.name,
+                overlay_peering_interface="Loopback0",
+            )
+
+            self.structured_config.router_bgp.neighbors.append(neighbor)
 
     def _set_mpls_mesh_pe(self: AvdStructuredConfigOverlayProtocol) -> None:
         if not self.shared_utils.overlay_mpls or not self.inputs.bgp_mesh_pes:
             return
 
         for fabric_switch in natural_sort(self.shared_utils.all_fabric_devices):
-            if self._mpls_route_reflectors is not None and fabric_switch in self._mpls_route_reflectors:
+            if fabric_switch in self._mpls_route_reflectors:
                 continue
             if fabric_switch == self.shared_utils.hostname:
                 continue
 
-            peer_facts = self.shared_utils.get_peer_facts(fabric_switch, required=True)
+            peer_facts = self.shared_utils.get_peer_facts(fabric_switch)
             if self._is_peer_mpls_client(peer_facts) is not True:
```

Comment:
to copy style in RR Peers
```suggestion
            if not self._is_peer_mpls_client(peer_facts):
```